src/promptflow/promptflow/_sdk/operations/_test_submitter.py

```diff
@@ -216,57 +204,57 @@ def change_logger_level(level):
             yield
             logger.setLevel(origin_level)
 
-        @contextlib.contextmanager
-        def add_prefix():
-            write = sys.stdout.write
-
-            def prefix_output(*args, **kwargs):
-                if args[0].strip():
-                    write(f"{Fore.LIGHTBLUE_EX}[{self.dataplane_flow.name}]: ")
-                write(*args, **kwargs)
-
-            sys.stdout.write = prefix_output
-            yield
-            sys.stdout.write = write
+        def show_node_log_and_output(node_run_infos, show_node_output):
+            """Show stdout and output of nodes."""
+            for node_name, node_result in node_run_infos.items():
+                # Prefix of node stdout is "%Y-%m-%dT%H:%M:%S%z"
+                pattern = r"\[\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}\+\d{4}\] "
+                node_logs = re.sub(pattern, "", node_result.logs["stdout"])
+                if node_logs:
+                    for log in node_logs.rstrip("\n").split("\n"):
+                        print(f"{Fore.LIGHTBLUE_EX}[{node_name}]:", end=" ")
+                        print(log)
+                if show_node_output:
+                    print(f"{Fore.CYAN}{node_name}: ", end="")
+                    # TODO executor return a type string of generator
+                    node_output = node_result.output
+                    if isinstance(node_result.output, GeneratorType):
+                        node_output = "".join(get_result_output(node_output))
+                    print(f"{Fore.LIGHTWHITE_EX}{node_output}")
 
         def print_chat_output(output):
-            chat_output = output
-            if isinstance(output, Iterable):
-                chat_output = ""
-                for event in output:
+            if isinstance(output, GeneratorType):
+                for event in get_result_output(output):
                     print(event, end="")
                     # For better animation effects
                     time.sleep(0.01)
-                    chat_output += event
                 # Print a new line at the end of the response
                 print()
             else:
-                print(chat_output)
-            return chat_output
-
-        def get_generator_values(generator):
-            if isinstance(generator, GeneratorType):
-                if generator not in generator_record:
-                    generator_record[generator] = list(generator)
-                return generator_record[generator]
-            else:
-                return generator
+                print(output)
+
+        def get_result_output(output):
+            if isinstance(output, GeneratorType):
+                if output in generator_record:
+                    output = iter(generator_record[output].items)
+                else:
+                    proxy = output.gi_frame.f_locals["proxy"]
```

Comment:
We may avoid using the internal field `gi_frame` of GeneratorType.

Reply:
Not found other better way to get the input value of generator.
And gi_frame is exposed by generator type.
https://docs.python.org/3.8/library/inspect.html